### PR TITLE
FEATURE: allows to unfollow direct message channel

### PR DESCRIPTION
**Pull request by jjaffeux**
Screenshot 2021-12-30 at 11 21 20

Kapture 2021-12-30 at 11 24 47

### CODE REVIEW

**Comment by markvanlan** [comment]
So if you unfollow a channel because you don't want it in the list, when that user messages you, you'll never see it until you manually refollow the channel? 

I think we should add a check in message creation, that checks if it's a DM channel, and ensures all members are following

**Comment by jjaffeux** [comment]
> So if you unfollow a channel because you don't want it in the list, when that user messages you, you'll never see it until you manually refollow the channel?
> 
> I think we should add a check in message creation, that checks if it's a DM channel, and ensures all members are following

Indeed good point thanks :+1:

**Comment by jjaffeux** [comment]
@markvanlan ok that should do it: https://github.com/discourse/discourse-chat/pull/402/commits/c171be04732dc0255590a0b69625e1993e73b2f0

**Comment by jjaffeux** [comment]
test failure is fixed in main

File: lib/direct_message_channel_creator.rb [hunk 1]
```diff
@@ -3,21 +3,35 @@
...
-        muted: false,
-        desktop_notification_level: UserChatChannelMembership::NOTIFICATION_LEVELS[:always],
-        mobile_notification_level: UserChatChannelMembership::NOTIFICATION_LEVELS[:always],
-      )
+  def self.create!(users)
+    direct_messages_channel = DirectMessageChannel.for_user_ids(users.map(&:id).uniq)
+    unless direct_messages_channel
+      direct_messages_channel = DirectMessageChannel.create!(users: users)
+      chat_channel = ChatChannel.create!(chatable: direct_messages_channel)
+    else
```
Comment (CvX) [inline lib/direct_message_channel_creator.rb:11]: v. minor thing: could you swap these around and make it and if-else?

**Review by CvX — APPROVED** [review]
Looks good! 🚢

File: lib/direct_message_channel_creator.rb (hunk 1 above)
Comment (jjaffeux) [reply lib/direct_message_channel_creator.rb:11]: yep sure

**Pull request merged**